### PR TITLE
BUG: Resolve regression on imagej TIFF hyperstacks

### PR DESCRIPTION
**Pull request by FirefoxMetzger**
Closes: #681 

This PR implements a fix that resolves the regression reported in #681 . It also cleans up the tifffile plugin a bit by adding the ability to read multiple volumes (if present in the data). 

It does not address the `imread` bug discussed in the same issue. That is, it still returns a single page of the TIFF instead of returning the first image.

@almarklein It appears that the `volread` bug is a regression introduced by https://github.com/imageio/imageio/issues/558 which attempts to fix https://github.com/imageio/imageio/pull/559. That PR didn't add a regression test, and the test image from #559 has disappeared, so it's not easy to guard against regression for that issue.

### CODE REVIEW

**Review by almarklein — APPROVED** [review]
I saw that there has been an extensive discussion on how Tiff works, and how we should interpret it in imageio. I've only superficially read through it. I'm just going to trust your judgement in this matter :)

**Comment by FirefoxMetzger** [comment]
I replaced the test that uses the original image (for which we are still waiting for copyright/licensing) with an artificial hyperstack that is filled with 0s. This reproduces the original problem and circumvents licensing issues.

**Comment by FirefoxMetzger** [comment]
@almarklein I created an entry for `tifffile` in our setup.py and updated our CI to install it when running the tests. Please check if you are okay with this.

The regression test in this PR relies on `tifffile` to create an ImageJ hyperstack (was broken if read via ImageIO). 

Given that our future plan is to remove our vendored tifffile, I am using this opportunity to start moving towards doing that :)

**Review by almarklein — APPROVED** [review]

File: tests/test_tifffile.py [hunk 1]
```diff
@@ -3,14 +3,15 @@
...
 import datetime
-
 import numpy as np
 
 from pytest import raises
 from imageio.testing import run_tests_if_main, get_test_dir, need_internet
 from imageio.core import get_remote_file
-
 import imageio
+import imageio as iio
```
Comment (mkcor) [inline tests/test_tifffile.py:12]: I would pick one...

File: tests/test_tifffile.py [hunk 2]
```diff
@@ -130,11 +131,17 @@ def test_tifffile_reading_writing():
         assert md["description"] == "another desc"
 
 
-# # commented out until we get permission to use this image
-# # in our test suite
-# def test_imagej_hyperstack():
-#     img = imageio.volread("NPCsingleNucleus.tif")
-#     assert img.shape == (15, 2, 180, 183)
+def test_imagej_hyperstack(tmp_path):
+    # artifical hyperstack
```
Comment (mkcor) [inline tests/test_tifffile.py:135]:
```suggestion
    # create artifical hyperstack
```

File: tests/test_tifffile.py (hunk 1 above)
Comment (FirefoxMetzger) [reply tests/test_tifffile.py:12]: > I would pick one...

Yep, I agree @mkcor and you are very correct in pointing this out.

I will move the entire test suite to `import imageio as iio` once I go through and do a test-styling PR. In the current suite, some files use `import imageio`, others use `import imageio as iio` and some use both. The change here actually doesn't worsen the situation overall 😅 .

There are a few other test-related things on my bucket list (serve standard images via a pytest-fixture, ensure that plugins are enforced for plugin-specific tests, etc.) so I think it makes sense to tread all of them in a dedicated PR that cleans up the test suite ... maybe we can even improve coverage in the process (wishful thinking™).

So yes, will be fixed. Unfortunately not in this PR, but in a future one instead. On the bright side, this PR will merge once the tests all pass :)

**Pull request merged**